llvm/lib/CodeGen/ReachingDefAnalysis.cpp

```diff
@@ -48,12 +54,28 @@ static bool isValidRegDefOf(const MachineOperand &MO, MCRegister PhysReg,
   return TRI->regsOverlap(MO.getReg(), PhysReg);
 }
 
+static bool isFIDef(const MachineInstr &MI, int FrameIndex,
+                    const TargetInstrInfo *TII) {
+  int DefFrameIndex = 0;
+  int SrcFrameIndex = 0;
+  if (TII->isStoreToStackSlot(MI, DefFrameIndex) ||
+      TII->isStackSlotCopy(MI, DefFrameIndex, SrcFrameIndex)) {
+    return DefFrameIndex == FrameIndex;
+  }
+  return false;
+}
+
 void ReachingDefAnalysis::enterBasicBlock(MachineBasicBlock *MBB) {
   unsigned MBBNumber = MBB->getNumber();
   assert(MBBNumber < MBBReachingDefs.numBlockIDs() &&
          "Unexpected basic block number.");
   MBBReachingDefs.startBasicBlock(MBBNumber, NumRegUnits);
 
+  MBBFrameObjsReachingDefs[MBBNumber].resize(NumStackObjects);
+  for (unsigned FOIdx = 0; FOIdx < NumStackObjects; ++FOIdx) {
+    MBBFrameObjsReachingDefs[MBBNumber][FOIdx].push_back(-1);
```

Comment:
Is this initialization needed? I think you could just check `MBBFrameObjsReachingDefs[MBBNumber][FrameIndex - ObjectIndexBegin].empty()` in `getReachingDef`. This change would reduce memory usage.

Reply:
Strictly speaking I don't think it is needed. But `-1` means something was defined before the begining of MBB (see comment before this line `MBBReachingDefs.append(MBBNumber, Unit, -1);`). I would prefer to leave this as is to stay consistent with registers.